bin/check-ts.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+# Script to run the type-checker in all directories in the npm workspaces list
+
+set -e
+
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+REPO="$SCRIPT_DIR/.."
+
+cd "$REPO"
+DIRS=$( jq --raw-output ".workspaces[]" package.json )
+
+for dir in $DIRS
+do
+    "$SCRIPT_DIR"/cd_sh "$dir" "[ ! -f eslint.config.js ] || npx tsc"
+done
```

Comment:
All this should be equivalent to calling `npm run build` from the root directory, no?

Reply:
Almost, but not quite. The issue is that `npm run build` does invoke `tsc` in each directory, but then `tsc` reports errors using paths that are relative to some directory below the one in which `npm run build` was executed, making finding the error locations hard for humans like me and impossible for tools. The thing this script does that `npm run build` does not is in `cd_sh`, which reports the usual entering/exiting directory messages before and after running commands in other directories, making it possible to jump to the errors reported by those commands.

Reply:
I would be happy to have a way to get npm or tsc to itself report errors using paths relative to the root, but could not find a way. I also didn't manage to insert a script wrapping tsc with directory reporting in a way that npm would actually call it.